rascal-lsp/pom.xml

```diff
@@ -112,6 +112,11 @@
       <artifactId>log4j-jul</artifactId>
       <version>${log4j2.version}</version>
     </dependency>
+    <dependency>
+      <groupId>org.eclipse.lsp4j</groupId>
+      <artifactId>org.eclipse.lsp4j.debug</artifactId>
+      <version>0.21.0</version>
```

Comment:
the versions for lsp4j.debug and lsp4j are evolving similarly, however, it would require to upgrade lsp4j version from 0.15 to 0.21 . Is there a way to test that the upgrade do not break anything ?

Reply:
These are the manual tests we do for every release:

- [ ] Open a Rascal REPL using the CMD+P start-rascal-terminal command (without a project)
- [ ] Manually create a new Rascal project, with src folder, META-INF/RASCAL.MF file with `Required-Libraries: |lib://rascal-lsp|`
- [ ] Can edit a new Rascal file in the Rascal project
- [ ] Save on Rascal file triggers type-checker, errors visible
- [ ] Rascal outline works
- [ ] Clicking links in REPL opens editors and websites
- [ ] `rascal>1 + 1` on the REPL
- [ ] `import IO; println("Hello Rascal!");`
- [ ] in editor, click on use of name jumps to definition
- [ ]  jump-to-definition also work to libraries and inside of libraries #150
- [ ] clicking in outline jumps to editor to right position
- [ ] syntax highlighting in editor works
- [ ] add dependency on another project by editing `RASCAL.MF`: `Require-Libraries: |lib://otherProject|`, import a module and test the type-checker as well as the interpreter for correct resolution, also test a new REPL and import a library modules from the imported library.
- [ ] Run `import demo::lang::pico::LanguageServer;` and  `demo::lang::pico::LanguageServer::main();` and test the editor of some example pico files:
   - [ ] pico syntax highlighting
   - [ ] pico parse errors
   - [ ] pico jump-to-definition
   - [ ] pico code lenses with command (Rename variables a to b)
- [ ] try `:edit demo::lang::pico::LanguageServer` #151

Reply:
Do we need to be on 0.21.0 of the debug protocol? In general we try to be the lowest we need, to support older version of VS Code.

Reply:
The 0.21 brings the 1.6 version support for DAP, if we stay in 0.15 of LSP4j.debug, we support 1.56 . It's not majors changes so it's probably not a problem to downgrade lsp4j.debug to the lsp4j version if it's preferable.